dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/DashbuilderMigrationService.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dashbuilder;
+
+import java.util.HashMap;
+import java.util.function.Predicate;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.backend.server.spaces.SpacesAPIImpl;
+import org.uberfire.commons.services.cdi.Startup;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.file.Path;
+import org.uberfire.java.nio.file.FileSystem;
+import org.uberfire.java.nio.file.SimpleFileVisitor;
+import org.uberfire.java.nio.file.attribute.BasicFileAttributes;
+import org.uberfire.java.nio.file.FileVisitResult;
+import org.uberfire.java.nio.file.Files;
+import org.uberfire.java.nio.IOException;
+import org.uberfire.spaces.SpacesAPI;
+
+import static org.kie.soup.commons.validation.PortablePreconditions.checkNotNull;
+
+@ApplicationScoped
+@Startup
+public class DashbuilderMigrationService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DashbuilderMigrationService.class);
+
+    private IOService ioService;
+    private FileSystem oldDatasetsFS;
+    private FileSystem datasetsFS;
+    private FileSystem pluginsFS;
+    private FileSystem perspectivesFS;
+    private FileSystem navigationFS;
+
+    public DashbuilderMigrationService() {
+
+    }
+
+    @Inject
+    public DashbuilderMigrationService(
+            final @Named("ioStrategy") IOService ioService,
+            final @Named("datasetsFS") FileSystem datasetsFS,
```

Comment:
Just to make sure, this datasetsFS points to the new dashbuilder structure right?

Reply:
yes
see SystemConfigProducer.java

Reply:
See [AF-2059](https://issues.jboss.org/browse/AF-2059) for task `Lock space on migration (HA enviroment)`